### ISSUE TITLE
[Feature Request]: ConditionBuilder prop to set filters which are not allowed to be removed by user since it has mandatory filters

### ISSUE THREAD

**Issue by jakubkloc** [carbon-design-system/ibm-products#9331]
### The problem

Currently, the ConditionBuilder component allows users to remove any condition, including fields that are mandatory (for example mandatory to create api query). This creates a poor user experience where users can inadvertently remove mandatory filter.

(this is basically usability issue but template for usability issue is focus on accessibility, so I raise it as new. feature request.)

### The solution

ConditionBuilder should support a configuration prop to pass specific fields as required/mandatory
Required filters should be automatically retained when users attempt to clear all filters

When this mandatory filter prop is passed to ConditionBuilder,  ConditionBuilder should display information (with message adjusted by developer in different prop) as warning tooltip or warning icon with tooltip rendered next to ConditionBuilder 

### Acceptance criteria

- Provide an onRemoveItem callback that is triggered before removing any condition, group, subgroup, or action, returning control (e.g., preventRemove) based on the provided context (type, item, group, state)
- Prevent removal when onRemoveItem returns preventRemove: true, allowing consumers to enforce rules such as protecting mandatory conditions (e.g., specific properties like region)
- Note: User can use description prop to render any message in tooltip on hover of s property
```
onRemoveItem={({ type, item, group, state }) => {
  if (type === 'condition' && item?.property === 'region') {
    return { preventRemove: true };
  }
  return { preventRemove: false };
}}
```

### Owner/main maintainer(s)

amal-k-joy

### Product/offering

FTM

### Business priority

High Priority = pressing release

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues
